applications/IgaApplication/custom_elements/iga_truss_element.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+//
+//  Main authors:   Anna Bauer
+//                  Thomas Oberbichler
+//                  Tobias Teschemacher
+*/
+
+// System includes
+#include "includes/define.h"
+#include "includes/variables.h"
+
+// External includes
+
+// Project includes
+#include "iga_truss_element.h"
+#include "iga_application_variables.h"
+
+namespace Kratos {
+
+Element::Pointer IgaTrussElement::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    auto geometry = GetGeometry().Create(ThisNodes);
+
+    return Kratos::make_shared<IgaTrussElement>(NewId, geometry,
+        pProperties);
+}
+
+void IgaTrussElement::GetDofList(
+    DofsVectorType& rElementalDofList,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    rElementalDofList.resize(NumberOfDofs());
+
+    for (std::size_t i = 0; i < NumberOfNodes(); i++) {
+        SetDof(rElementalDofList, i, 0, DISPLACEMENT_X);
+        SetDof(rElementalDofList, i, 1, DISPLACEMENT_Y);
+        SetDof(rElementalDofList, i, 2, DISPLACEMENT_Z);
+    }
+
+    KRATOS_CATCH("")
+}
+
+void IgaTrussElement::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    rResult.resize(NumberOfDofs());
+
+    for (std::size_t i = 0; i < NumberOfNodes(); i++) {
+        SetEquationId(rResult, i, 0, DISPLACEMENT_X);
+        SetEquationId(rResult, i, 1, DISPLACEMENT_Y);
+        SetEquationId(rResult, i, 2, DISPLACEMENT_Z);
+    }
+
+    KRATOS_CATCH("")
+}
+
+void IgaTrussElement::Initialize()
+{
+    mReferenceBaseVector = GetActualBaseVector();
+}
+
+IgaTrussElement::Vector3D IgaTrussElement::GetActualBaseVector() const
```

Comment:
What is the actual BaseVector? Is there also a fake one? :)

If it is the current configuration then I think this would be a better name
and why no function abt getting the ref-configuration?

Reply:
I just want to use the same names as in the `Brep...` banch.

> If it is the current configuration then I think this would be a better name

Agree

> why no function abt getting the ref-configuration?

I think because it is called inside initialize when act = ref